lib/sequent/core/aggregate_root.rb

```diff
@@ -80,6 +71,35 @@ def load_from_history(stream, events)
         events.each { |event| apply_event(event) }
       end
 
+      def initialize_for_streaming(stream)
+        @uncommitted_events = []
+        @sequence_number = 1
+        @event_stream = stream
+      end
+
+      def stream_from_history(stream_events)
+        _stream, events = stream_events
+        fail 'Empty history' if events.empty?
+
+        @id ||= events.first.aggregate_id
+        events.each { |event| apply_event(event) }
+      end
+
+      def self.stream_from_history(stream)
+        aggregate_root = allocate
+        aggregate_root.initialize_for_streaming(stream)
+        aggregate_root
+      end
+
+      def initialize(id)
```

Comment:
Can you move the `initialize` back to its original position? I think it is good to have that as first instance method and reduces the number of changes.